solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -865,6 +871,76 @@ public void removeLiveNodesListener(LiveNodesListener listener) {
     liveNodesListeners.remove(listener);
   }
 
+  /**
+   * Returns the lowest Solr version among all live nodes in the cluster.
+   *
+   * @return the lowest Solr version as a String, or null if no version is found
+   * @throws KeeperException if a ZooKeeper error occurs
+   * @throws InterruptedException if the thread is interrupted
+   */
+  public String getLowestSolrVersion() throws KeeperException, InterruptedException {
+    List<String> liveNodeNames = zkClient.getChildren(LIVE_NODES_ZKNODE, null, true);
+    String lowestVersion = null;
+
+    for (String nodeName : liveNodeNames) {
+      String path = LIVE_NODES_ZKNODE + "/" + nodeName;
+      byte[] data = zkClient.getData(path, null, null, true);
+      if (data == null || data.length == 0) {
+        continue;
```

Comment:
if there's no data, this means it's a Solr 9.8 node (or earlier).  This is important!